Add zfs cloning

wishdev opened the pull request:
Adds the ability to clone a snapshot when creating a zfs volume

gregdek commented:
@johanwiren Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please give it a +1; at that point, we will merge.

johanwiren commented:
I think that the "clone" option should be named "origin" to better follow zfs naming.

Also, please add an example to the doc in how to create and destroy clones.

johanwiren commented on system/zfs.py:
```diff
@@ -66,6 +66,10 @@
+  clone:
+    description:
+      - Name of the snapshot to clone
+   required: False
```
Wrong indentation.

wishdev commented:
@johanwiren - Switched the property name, cleaned up the identing issue and added examples. I only added a generic example for destroying a file system because there is nothing special about removing a cloned file system.

johanwiren commented:
Great work!

@gregdek I approve.
